lib/posix_channel.ml

```diff
@@ -190,8 +190,14 @@ let receive protocols =
   | V4V_proxy(_, _) -> assert false (* weight is 0 above *)
   | TCP_proxy(ip, port) ->
     let s = Unix.socket Unix.PF_INET Unix.SOCK_STREAM 0 in
-    Unix.connect s (Unix.ADDR_INET(Unix.inet_addr_of_string ip, port));
-    s
+    begin
+      try
+        Unix.connect s (Unix.ADDR_INET(Unix.inet_addr_of_string ip, port));
+        s
+      with e ->
+        Unix.close s;
+        raise e
```

Comment:
We should probably wrap this into an `Internal_error` and raise that instead.

Reply:
Raise the native exception aims to keep consistent with https://github.com/xapi-project/xcp-idl/blob/master/lib/posix_channel.ml#L197-L203